File: leetcode/240. Search a 2D Matrix II.kt
class Solution {
    fun searchMatrix(matrix: Array<IntArray>, target: Int): Boolean {
        var i = 0
        var j = matrix[0].size - 1

        while (i < matrix.size && j >= 0) {
            if (matrix[i][j] == target) return true
            if (matrix[i][j] < target) {
                i++
            } else {
                j--
            }
        }
        return false
    }
}

File: leetcode/240. Search a 2D Matrix II.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        m = len(matrix)
        n = len(matrix[0])
        
        i = 0
        j = n-1
        while i < m and j >= 0:
            if matrix[i][j] == target:
                return True
            elif matrix[i][j] < target:
                i += 1
            else:
                j -= 1
        return False
